Remove redundant sprintf argument.

Aliance opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #22820
| License       | MIT
| Doc PR        | –

cc @fabpot @ogizanagi

ro0NL commented on src/Symfony/Bridge/Twig/Tests/Node/TransNodeTest.php:
```diff
@@ -53,6 +53,6 @@ protected function getVariableGetterWithStrictCheck($name)
         }
 
-        return sprintf('($context["%s"] ?? $this->getContext($context, "%s"))', $name, $name, $name);
+        return sprintf('($context["%s"] ?? $this->getContext($context, "%s"))', $name, $name);
```
`%1$s`? thus assign `$name` once here as well.

javiereguiluz replied:
Don't you find that notation (`%1$s`) difficult to understand? I always struggle with it.

ro0NL replied:
No strong opinion. Just to make things less error prone :) basically the two assignments must always stay the same, thus id prefer a single assignment.

Aliance replied:
@ro0NL I thought about it, but decided not to change. But I agree that `%1$s` is better.

chalasr replied:
I prefer `%1$s`. That's how `sprintf()` should be used to avoid duplications, and we already have some usages in the code base

Tobion commented:
This PR should be opened against 2.7 as it is wrong there as well.

Aliance commented:
@Tobion there is different code in different branches:
https://github.com/symfony/symfony/blob/v2.7.35/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L74-L85

https://github.com/symfony/symfony/blob/v2.8.28/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L74-L85

https://github.com/symfony/symfony/blob/v3.0.9/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L67-L70

https://github.com/symfony/symfony/blob/v3.1.10/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L67-L74

https://github.com/symfony/symfony/blob/v3.2.13/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L74-L81

https://github.com/symfony/symfony/blob/v3.3.10/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L74-L81

https://github.com/symfony/symfony/blob/v3.4.0-BETA1/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L103-L110

https://github.com/symfony/symfony/blob/v4.0.0-BETA1/src/Symfony/Bridge/Twig/Tests/Node/FormThemeTest.php#L103-L106

So in some branches there is php version checks, in others – not. May be merge this pr to branches w/o php version checks and create another one for other branches?

Tobion commented:
@Aliance it's enough if you open a PR for 2.7. We then merge it upwards and will resolve conflicts.

Aliance committed:
Remove redundant sprintf arguments.

Aliance commented:
@Tobion I rebased it on 2.7, also changed duplicated arguments to `%1$s` and also tried to find all the same occurrences with regexp `sprintf\('(.*?)', \$(.*?)(, \$\2\)*(, (.*?)))`

Aliance commented:
Does anyone knows what's wrong with the appveyor tests? I failed to find any errors...

ro0NL commented on src/Symfony/Component/DomCrawler/Form.php:
```diff
@@ -437,7 +437,7 @@ private function initialize()
             $formId = Crawler::xpathLiteral($this->node->getAttribute('id'));
 
-            $fieldNodes = $xpath->query(sprintf('descendant::input[@form=%s] | descendant::button[@form=%s] | descendant::textarea[@form=%s] | descendant::select[@form=%s] | //form[@id=%s]//input[not(@form)] | //form[@id=%s]//button[not(@form)] | //form[@id=%s]//textarea[not(@form)] | //form[@id=%s]//select[not(@form)]', $formId, $formId, $formId, $formId, $formId, $formId, $formId, $formId));
+            $fieldNodes = $xpath->query(sprintf('descendant::input[@form=%s] | descendant::button[@form=%1$s] | descendant::textarea[@form=%1$s] | descendant::select[@form=%1$s] | //form[@id=%1$s]//input[not(@form)] | //form[@id=%1$s]//button[not(@form)] | //form[@id=%1$s]//textarea[not(@form)] | //form[@id=%1$s]//select[not(@form)]', $formId));
```
lol :) definitely up for `%1$s` now :)

Tobion commented:
Good catch, thanks @Aliance.
